### PR TITLE
Fix Broken Links and Permanent Redirects

### PR DESCRIPTION
**Pull request by yellowjaguar5**
I ran `make linkcheck` on the `master` branch and I noticed that there were several links that were broken or had permanent redirects. This PR attempts to fix all broken links and permanent redirects, as shown in `make linkcheck`.

- [x] Fix all unreachable page errors (40X)
- [x] Fix all permenent redirects
- [x] Fix Anchor not found errors
- [x] Ensure every link that can be `https` is `https`

Edit: **I'll be away from the internet for two weeks (back on August 11th) and I plan to continue working on this PR when I return.**

### CODE REVIEW

**Comment by AustinShalit** [comment]
Please check all links to see if there is a HTTS version available.

**Comment by Daltz333** [comment]
In regards to the anchors, I'm okay with 
```
linkcheck_anchors = False
```
being set in the conf.py

This disables checking the anchors. We simply can't trust how websites javascript messes with direction. This flag, when disabled, also greatly speeds up the build.

File: source/conf.py [hunk 1]
```diff
@@ -113,3 +112,8 @@ def setup(app):
 suppress_warnings = ['epub.unknown_project_files']
 
 sphinx_tabs_valid_builders = ['epub']
+
+
+# -- Options for the linkcheck builder ---------------------------------------
```
Comment (Daltz333) [inline source/conf.py]: Revert for now, this can be done in the other PR

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 2]
```diff
@@ -36,14 +36,14 @@ On-shaft encoders
 
 .. todo:: add picture
 
-On-shaft encoders (such as the `AMT103-V <https://www.digikey.com/product-detail/en/cui-inc/AMT103-V/102-1308-ND/827016>`__ available through FIRST Choice) couple to a shaft by fitting *around* it, forming a friction coupling between the shaft and a rotating hub inside the encoder.
+On-shaft encoders (such as the `AMT103-V <https://www.cui.com/product/motion/rotary-encoders/incremental/modular/amt10-series>`__ available through FIRST Choice) couple to a shaft by fitting *around* it, forming a friction coupling between the shaft and a rotating hub inside the encoder.
```
Comment (Daltz333) [inline source/docs/hardware/sensors/encoders-hardware.rst]: Why this change? That is not the AMT103-V

File: source/docs/software/getting-started/offline-installation-preparations.rst [hunk 3]
```diff
@@ -15,8 +15,8 @@ All Teams
 ^^^^^^^^^
 
 -  `2019 NI Update Suite <https://www.ni.com/download/first-robotics-software-2017/7904/en/>`__ (Note: Requires decryption key from kickoff broadcast!)
--  `2019 FRC Radio Configuration Utility <https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1.zip>`__ or `2019 FRC Radio Configuration Utility Israel Version <https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1_IL.zip>`__
--  (Optional - Veterans Only!) `Classmate/Acer PC Image <https://frc-events.firstinspires.org/services/DSImages/2019>`__
+-  `2019 FRC Radio Configuration Utility <https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1.zip>`__ or `2019 FRC Radio Configuration Utility Israel Version <http://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1_IL.zip>`__
```
Comment (Daltz333) [inline source/docs/software/getting-started/offline-installation-preparations.rst]: Why no SSL? The SSL link is valid.

File: source/docs/software/getting-started/radio-programming.rst [hunk 4]
```diff
@@ -68,7 +68,7 @@ Download the latest FRC Radio Configuration Utility Installer from the following
 
 `FRC Radio Configuration 19.1.1 <https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1.zip>`_
 
-`FRC Radio Configuration 19.1.1 Israel Version <https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1_IL.zip>`_
+`FRC Radio Configuration 19.1.1 Israel Version <http://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1_IL.zip>`_
```
Comment (Daltz333) [inline source/docs/software/getting-started/radio-programming.rst]: No SSL?

File: source/docs/hardware/sensors/encoders-hardware.rst (hunk 2 above)
Comment (yellowjaguar5) [reply source/docs/hardware/sensors/encoders-hardware.rst]: Linkcheck was having problems with the [digikey link](https://www.digikey.com/product-detail/en/cui-inc/AMT103-V/102-1308-ND/827016), so I wanted to go straight to the manufacturer. I'll swap it for a link to the [AMT103-V](https://www.cui.com/product/motion/rotary-encoders/incremental/modular/amt10-v-kit/amt103-v)

File: source/docs/software/getting-started/offline-installation-preparations.rst (hunk 3 above)
Comment (yellowjaguar5) [reply source/docs/software/getting-started/offline-installation-preparations.rst]: Swapped for SSL.

File: source/docs/software/getting-started/radio-programming.rst (hunk 4 above)
Comment (yellowjaguar5) [reply source/docs/software/getting-started/radio-programming.rst]: Swapped for SSL.

**Comment by yellowjaguar5** [comment]
1. Changes to layout.html were not intended and they were reverted.
2. yay!
3. I'm not sure why I removed SSL from some links and I readded SSL where it was removed.

**Comment by yellowjaguar5** [comment]
@Daltz333 Please review.
